### PR TITLE
Fix/not null field aggregations

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

This PR splits the aggregation base types (e.g. `StringAggregateSelection`) into `Nullable` and `NonNullable` versions, depending on the base type (e.g. String or String!) a more accurate type will be selected, allowing for nullable fields to be used in aggregations. The related issue has an example of this issue

## Other changes
 
* Aggregation types extracted from `make-augmented-schema` and moved to a new class `AggregationTypesMapper` to simplify the process of getting the correct type
* Several schema tests have been updated

# Issue

#594

### CODE REVIEW

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Just one comment! Then a bit of a nasty merge to be done!

File: packages/graphql/src/schema/aggregations/aggregation-types-mapper.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ObjectTypeComposer, SchemaComposer } from "graphql-compose";
+import { numericalResolver, idResolver } from "../resolvers";
+
+export class AggregationTypesMapper {
+    private requiredAggregationSelectionTypes: Record<string, ObjectTypeComposer<unknown, unknown>>;
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/schema/aggregations/aggregation-types-mapper.ts:24]: Well, we already use required (e.g. `!field.typeMeta.required`) and 2 variables `nullableAggregationSelectionTypes` and `nonNullableAggregationSelectionTypes` can easily lead to confusion IMO

Happy to change if it is a deal breaker

[the remark this answers is not in the document]
Comment (darrellwarde) [reply packages/graphql/src/schema/aggregations/aggregation-types-mapper.ts:24]: Meeeehhhhh, leave it be!

**Review by darrellwarde — APPROVED** [review]
LGTM! 🙂

**Pull request merged**